src/internal/packager/images/pull.go

```diff
@@ -7,506 +7,393 @@ package images
-			lines = append(lines, fmt.Sprintf("image - %s@%s with platform %s", name, desc.Digest.String(), desc.Platform.String()))
-		}
-		imageOptions := strings.Join(lines, "\n")
-		return fmt.Errorf("%s resolved to an OCI image index which is not supported by Zarf, select a specific platform to use: %s", refInfo.Reference, imageOptions)
-	}
-	return nil
+type imagePullInfo struct {
+	registryOverrideRef string
+	ref                 string
+	manifestDesc        ocispec.Descriptor
+	byteSize            int64
 }
 
-func getDockerEndpointHost() (string, error) {
-	dockerCli, err := command.NewDockerCli(command.WithStandardStreams())
-	if err != nil {
-		return "", err
-	}
-	newClientOpts := flags.NewClientOptions()
-	err = dockerCli.Initialize(newClientOpts)
-	if err != nil {
-		return "", err
-	}
-	store := dockerCli.ContextStore()
-	metadata, err := store.GetMetadata(dockerCli.CurrentContext())
-	if err != nil {
-		return "", err
-	}
-	endpoint, err := docker.EndpointFromContext(metadata)
-	if err != nil {
-		return "", err
-	}
-	return endpoint.Host, nil
+type imageDaemonPullInfo struct {
+	registryOverrideRef string
+	image               transform.Image
 }
 
 // Pull pulls all images from the given config.
-func Pull(ctx context.Context, cfg PullConfig) (map[transform.Image]v1.Image, error) {
+func Pull(ctx context.Context, cfg PullConfig) (map[transform.Image]ocispec.Manifest, error) {
 	l := logger.From(ctx)
-	var longer string
 	pullStart := time.Now()
 
 	imageCount := len(cfg.ImageList)
-	// Give some additional user feedback on larger image sets
-	if imageCount > 15 {
-		longer = "This step may take a couple of minutes to complete."
-	} else if imageCount > 5 {
-		longer = "This step may take several seconds to complete."
-	}
-
 	if err := helpers.CreateDirectory(cfg.DestinationDirectory, helpers.ReadExecuteAllWriteUser); err != nil {
 		return nil, fmt.Errorf("failed to create image path %s: %w", cfg.DestinationDirectory, err)
 	}
 
-	cranePath, err := clayout.Write(cfg.DestinationDirectory, empty.Index)
-	if err != nil {
-		return nil, err
-	}
-
-	// Give some additional user feedback on larger image sets
 	imageFetchStart := time.Now()
-	// TODO(mkcp): Remove message on logger release
-	spinner := message.NewProgressSpinner("Fetching info for %d images. %s", imageCount, longer)
-	defer spinner.Stop()
 	l.Info("fetching info for images", "count", imageCount, "destination", cfg.DestinationDirectory)
-
-	logs.Warn.SetOutput(&message.DebugWriter{})
-	logs.Progress.SetOutput(&message.DebugWriter{})
-
+	storeOpts := credentials.StoreOptions{}
+	credStore, err := credentials.NewStoreFromDocker(storeOpts)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get credentials: %w", err)
+	}
+	client := &auth.Client{
+		Client:     retry.DefaultClient,
+		Cache:      auth.NewCache(),
+		Credential: credentials.Credential(credStore),
+	}
+	platform := &ocispec.Platform{
+		Architecture: cfg.Arch,
+		OS:           "linux",
```

Comment:
Add a TODO here, we'll likely want to expose this as a configuration option to allow non-linux images to be pulled sometime in the future.